appcues/src/main/java/com/appcues/monitor/AppcuesActivityMonitor.kt

```diff
@@ -42,15 +46,32 @@ internal object AppcuesActivityMonitor : Application.ActivityLifecycleCallbacks
 
     override fun onActivityResumed(activity: Activity) {
         _isPaused = false
-        if (this.activity != activity) {
-            this.activityWeakReference = WeakReference(activity)
 
+        if (this.activity == null) {
+            this.activityWeakReference = WeakReference(activity)
             // notifies all subscribers that activity has changed
             activityMonitorListener.forEach { it.onActivityChanged(activity) }
         }
+
+        if (notifyConfigurationChanged) {
+            // notifies all subscribers that activity has changed
+            activityMonitorListener.forEach { it.onConfigurationChanged(activity) }
+            notifyConfigurationChanged = false
+        }
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        // whenever we hit onActivityCreated + savedInstanceState is not null
+        // it means configuration has changed (same activity was re-created
+        if (savedInstanceState != null) {
+            notifyConfigurationChanged = true
```

Comment:
this is the only part I really don't know definitively enough about to know if this is always going to work, hopefully so. I wish there were some more direct lifecycle callback specifically for configuration change, but maybe that just doesn't exist at the global lifecycle callback listener level.

Reply:
there is actually a method inside that interface called:
```kotlin

        /**
         * Called when the Activity configuration was changed.
         * @hide
         */
        default void onActivityConfigurationChanged(@NonNull Activity activity) {
        }
```
        
        but I'm not able to override it for some reason